libraries/AP_Logger/AP_Logger_MAVLink.cpp

```diff
@@ -354,10 +352,7 @@ void AP_Logger_MAVLink::Write_DMS(AP_Logger_MAVLink &logger_mav)
 
 void AP_Logger_MAVLink::stats_log()
 {
-    if (!_initialised) {
-        return;
-    }
-    if (stats.collection_count == 0) {
+    if (!_initialised || stats.collection_count == 0) {
         return;
```

Comment:
Please don't consolidate this

We end up with clauses which are ridiculously difficult to work out what they do.